src/Persistence/Proxy.php

```diff
@@ -14,6 +14,8 @@
 /**
  * @template TProxiedObject of object
  *
+ * @mixin TProxiedObject
```

Comment:
if all type hints are modified from `Zenstruck\Foundry\Proxy` to  `Zenstruck\Foundry\Persistence\Proxy` in user land, we must add this `@mixin` as well as all other deprecated methods. Otherwise SCA reports tons of errors